spec/features/idv/steps/profile_step_spec.rb

```diff
@@ -0,0 +1,64 @@
+require 'rails_helper'
+
+feature 'idv profile step', :idv_job do
+  include IdvStepHelper
+
+  context 'with valid information' do
+    it 'allows the user to continue to the address step' do
+      start_idv_from_sp
+      complete_idv_steps_before_profile_step
+      fill_out_idv_form_ok
+      click_idv_continue
+
+      expect(page).to have_content(t('idv.titles.select_verification'))
+      expect(page).to have_current_path(verify_address_path)
+    end
+  end
+
+  context 'after submitting valid information' do
+    it 'is not re-entrant' do
+      start_idv_from_sp
+      complete_idv_steps_before_profile_step
+      fill_out_idv_form_ok
+      click_idv_continue
+
+      # Attempt to go back to profile step
+      visit verify_session_path
+
+      # Get redirected to the address step
+      expect(page).to have_content(t('idv.titles.select_verification'))
+      expect(page).to have_current_path(verify_address_path)
+    end
+  end
+
+  it 'does not allow the user to advance without completing' do
+    start_idv_from_sp
+    complete_idv_steps_before_profile_step
+
+    # Try to skip ahead to address step
+    visit verify_address_path
+
+    # Get redirect to the profile step
+    expect(page).to have_current_path(verify_session_path)
+  end
```

Comment:
Can we combine all of these into one test? I feel like we are doing a lot of repetition with all of these IdV specs, which is slowing down the test suite a lot. What do you think about having one big test that goes through an end-to-end flow but also does other things that we can test in the same session?

For example, we can have a test where the user comes from an SP, then tries to skip ahead to the address form, gets redirected back to the Profile form, they fill it out with bad information, we verify the error messages, then they fill it out with correct info, then they try to go back to the Profile form, we verify they get redirected back to the verify address page, etc..

Reply:
Yes, some of these may be able to be merged, but I am wary of pulling all of them together, since when we do that it gets very hard to tell which features we are testing where. Then we risk ending up in the situation we were in before where it is hard to determine where we have coverage, and it is hard to change the tests when you want to make a change, because you can't determine what is being tested where.

Reply:
That's fine for now, but perhaps we need to have a broader team discussion about how we test the app. Do people feel we have too many integration tests?

Reply:
Yes, @jmhooper and I have chatted about this a bit. There are a lot of feature specs and they get increasingly slower as the number of pages increases because we navigate the entire flow to the page under test each time. Ideally, we can create the state we need for each page, navigate directly to that page and do the detailed tests without navigating over and over again. Then we can limit what is tested when testing the flow through the entire app.

Reply:
@monfresh @davemcorwin : I did end up combining the profile steps here. Wanna take another look, or think it's still good to go?